test/FunctionalTests/Tests/DataServices/UnitTests/ClientCSharpUnitTests/DataWebClientCSharp/LinqTests.cs

```diff
@@ -8154,6 +8154,34 @@ public void QueryRowCount_UriBuilder()
             }
         }
 
+        [TestMethod]
+        public void VerifyCountIsNotCarriedOver()
+        {
+            string expectedLongCountFunctionUri = "/service.svc/Leagues/LongCountFunction(Param1='Param1',Param2='Param2')?$orderby=ID";
+            string expectedCountFunctionUri = "/service.svc/Leagues/CountFunction(Param1='Param1',Param2='Param2')?$orderby=ID";
+
+            var countQuery = context.CreateQuery<League>("Leagues");
+            var longcountQuery = context.CreateQuery<League>("Leagues");
+
+            var countFunctionQuery = countQuery.CreateFunctionQuery<League>("CountFunction", true, new UriOperationParameter("Param1", "Param1"),
+                    new UriOperationParameter("Param2", "Param2"));
+            var longCountFunctionQuery = longcountQuery.CreateFunctionQuery<League>("LongCountFunction", true, new UriOperationParameter("Param1", "Param1"),
+                    new UriOperationParameter("Param2", "Param2"));
+            try
+            {
+                //check for both count an longcount
+                countFunctionQuery.Count();
+                longCountFunctionQuery.LongCount();
+
+            }

```

Comment:
Use Assert.Throws<T> , don't use the try/catch

Reply:
@xuzhg I could not figure out how Assert.Throws could be used in this case since the test seeks to verify the url formed although an exception is expected to be thrown since the Service does not exist. `Count()` and `LongCount()` unfortunately are executed when a service is being called.

Reply:
@paulodero for a unit test you should not even be making an actual network call. You should be able to fix that by using the `OnMessageCreating` client hook to return a fake response message and avoid the n/w call altogether:

```c#
    // create a custom request message
    class CustomReqMsg : HttpWebRequestMessage
    {
        public CustomReqMsg(DataServiceClientRequestMessageArgs args) : base(args) { }

        public override IODataResponseMessage GetResponse()
        {
            return new HttpWebResponseMessage(
                headers: new Dictionary<string, string>(),
                statusCode: 200,
                getResponseStream: () => new MemoryStream(Encoding.UTF8.GetBytes("2"))  // mimic a .Count response of 2
                );
        }
    }
   ...
   // use the custom request message to override OnMessageCreating
   context.Configurations.RequestPipeline.OnMessageCreating = args => new CustomReqMsg(args);
   // now when you make your `.Count` and `.LongCount` calls they shouldnt hit the n/w
```

Reply:
Better option. On it.

Reply:
Perfect solution! Done!